Mathlib/Combinatorics/SimpleGraph/Hasse.lean

```diff
@@ -156,7 +156,7 @@ def Walk.ofPathGraphHom {α} (G : SimpleGraph α) {n : ℕ} (hom : pathGraph (n
     exact Walk.cons hGadj w
 
 /-- Given a walk get a homomrfism from a pathGraph -/
-def Walk.toPathGraphHom {α} (G : SimpleGraph α) :
+theorem Walk.toPathGraphHom (G : SimpleGraph α) :
     ∀ {u v : α} (w : G.Walk u v), ∃(hom : pathGraph (w.length + 1) →g G), hom ⊥ = v ∧ hom ⊤ = u
```

Comment:
Are you not managing to define it?

Reply:
I wanted to have the homomorphism and a prof that hom ⊥ = v ∧ hom ⊤ for the recursive case so I used the ∃

Reply:
Do
```suggestion
def Walk.toPathGraphHomAux (G : SimpleGraph α) :
    ∀ {u v : α} (w : G.Walk u v), Σ' (hom : pathGraph (w.length + 1) →g G), hom ⊥ = v ∧ hom ⊤ = u
```
then 😉